shaarli2twitter/shaarli2twitter.php

```diff
@@ -86,12 +104,27 @@ function hook_shaarli2twitter_save_link($data, $conf)
     // We will use an array to generate hashtags, then restore original shaare tags.
     $data['tags'] = array_values(array_filter(explode(' ', $data['tags'])));
     for ($i = 0; $i < count($data['tags']); $i++) {
-        $data['tags'][$i] = '#'. $data['tags'][$i];
+         // Keep tags strictly alphanumerical because Twitter only allows that.
+        $data['tags'][$i] = get_tagify($data['tags'][$i]);
     }
 
+
     $data['permalink'] = index_url($_SERVER) . '?' . $data['shorturl'];
+   
+    // In case of note, we use permalink
+   if (is_link_note($data)) {
+        $data['url'] = $data['permalink'];
+        // Hide URL when sharing a note (microblog mode)
+        $hide = $conf->get('plugins.TWITTER_HIDE_URL', TWEET_HIDE_URL);
+        if ($hide == 'yes') {
+            $format = $conf->get('plugins.TWITTER_TWEET_FORMAT', TWEET_DEFAULT_FORMAT);
+            $data['url'] = '';
+            $tweet = format_tweet($data, $format);
+            $data['url'] = (get_current_length($tweet) >= TWEET_LENGTH) ? $data['permalink'] : '';
```

Comment:
Does this mean that the URL is hidden only if it's not too long? This doesn't really reflect what the setting is supposed to be. Also, calling `format_tweet()` twice doesn't seem really elegant. IMO, except if I misunderstood, those two lines should be removed.